ssas/service/main/main_test.go

```diff
@@ -28,6 +33,43 @@ func (s *MainTestSuite) TestResetSecret() {
 	assert.NotEqual(s.T(), "", output)
 }
 
+func (s *MainTestSuite) TestGetXDataWithClientID() {
+	creds, _ := s.createACOData()
+
+	output := captureOutput(func() { getXData(creds.ClientID, "") })
+	assert.Equal(s.T(), "fake x_data\n", output)
+}
+
+func (s *MainTestSuite) TestGetXDataWithAPIKey() {
+	creds, _ := s.createACOData()
+
+	// Build encoded api key to mimic auth header
+	APIKey := "Basic " + base64.StdEncoding.EncodeToString([]byte(creds.ClientID+":"+creds.ClientSecret))
+
+	output := captureOutput(func() { getXData("", APIKey) })
+	assert.Equal(s.T(), "fake x_data\n", output)
+}
+
+func (s *MainTestSuite) createACOData() (creds ssas.Credentials, group ssas.Group) {
```

Comment:
Would it make sense to move `createACOData()` (and `createIntrospectData()`) into the testutils.go file?

We can update the signature to accept the *testing.T object so you can still use the same assertions.